File: python/partition.py
from functools import lru_cache

@lru_cache(maxsize=None)
def partition(x: int):
    """
    returns the partition of integer 'x'
    """
    return sum([partition_x_base_n(x,i+1) for i in range(x)])

@lru_cache(maxsize=None)
def partition_x_base_n(x: int,n: int):
    """
    returns the number of ways to partition 'x' into 'n' integers greater than 0
    """
    if x == n: return 1
    if x-n <= n: return partition(x-n)
    return sum([partition_x_base_n(x-n,i+1) for i in range(n)])

File: java/Partition.java
class Partition{
    static int partition(int x){
        /* 
        returns the partition of integer 'x'
         */
        int ret = 0;
        for(int i = 1; i <= x; i++) ret += partitionXBaseN(x, i);
        return ret;
    }
    static int partitionXBaseN(int x, int n){
        /* 
        returns the number of ways to partition 'x' into 'n' integers greater than 0
         */
        if(x == n) return 1;
        if(x-n <= n) return partition(x-n);
        int ret = 0;
        for(int i = 1; i <= n; i++)ret += partitionXBaseN(x-n, i);
        return ret;
    }
}